File: frontend/src/types/collection.ts
/**
 * Tipos TypeScript para colecciones (espejo del backend).
 */

/** Colección de worldbuilding del usuario. */
export interface Collection {
  id: string;
  name: string;
  description: string;
  owner_id: string | null;
  created_at: string;
  updated_at: string | null;
  document_count: number;
  entity_count: number;
}

/** Payload para crear una colección. */
export interface CreateCollectionRequest {
  name: string;
  description: string;
}

/** Payload para actualizar una colección (todos los campos opcionales). */
export interface UpdateCollectionRequest {
  name?: string;
  description?: string;
}

import type { PaginatedMeta } from "./content";

/** Respuesta paginada de colecciones. */
export interface CollectionListResponse {
  data: Collection[];
  meta: PaginatedMeta;
}

File: backend/app/models/schemas/collection.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class CreateCollectionRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    description: str = Field(default="", max_length=2000)


class UpdateCollectionRequest(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = Field(default=None, max_length=2000)


class CollectionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    description: str
    owner_id: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    document_count: int = 0
    entity_count: int = 0
